src/core/IconButton/style.ts

```diff
@@ -0,0 +1,109 @@
+import { css, SerializedStyles } from "@emotion/react";
+import styled from "@emotion/styled";
+import { IconButton } from "@material-ui/core";
+import { getColors, getSpacings, Props } from "../styles";
+
+export interface ExtraProps extends Props {
+  active?: boolean;
+  color?: "primary" | "secondary";
+  disabled?: boolean;
+  size?: "small" | "medium";
+}
+
+const isActive = (props: ExtraProps): SerializedStyles => {
+  const { color } = props;
+  const colors = getColors(props);
+
+  return css`
+    color: ${color === "primary" ? colors?.primary[600] : colors?.primary[500]};
+
+    &:hover {
+      color: ${colors?.primary[600]};
+    }
+  `;
+};
+
+const isDisabled = (props: ExtraProps): SerializedStyles => {
+  const colors = getColors(props);
+
+  return css`
+    color: ${colors?.gray[300]};
+  `;
+};
+
+const primary = (props: ExtraProps): SerializedStyles => {
+  const colors = getColors(props);
+
+  return css`
+    color: ${colors?.primary[400]};
+
+    &:hover {
+      background: ${colors?.gray[300]};
+    }
+
+    &:active {
+      color: ${colors?.primary[600]};
+    }
+  `;
+};
+
+const secondary = (props: ExtraProps): SerializedStyles => {
+  const colors = getColors(props);
+
+  return css`
+    color: ${colors?.gray[500]};
+
+    &:hover {
+      background: none;
+      color: ${colors?.primary[400]};
+    }
+  `;
+};
+
+const small = (props: ExtraProps): SerializedStyles => {
+  const spacings = getSpacings(props);
+
+  return css`
+    &:hover {
+      background: none;
+    }
+
+    .MuiSvgIcon-root {
+      height: ${spacings?.l}px;
+      width: ${spacings?.l}px;
```

Comment:
The theme doesn't seem to have a (icon) "sizes" setting, so I used the spacing variable instead, but not sure if this is too unintuitive?

Reply:
Ooh this is a great point! @Janeece @happyimadesignr maybe we should define dimensions for the sizes in theme? Is it always gonna follow `spacings.l` and `spacings.s` for example?

Reply:
@tihuan @j-x-han definitely agree on defining icon sizes in the theme if we arent already -- we actually already have these sizes defined at the top of  the [Icons/Illustrations page](https://www.figma.com/proto/EaRifXLFs54XTjO1Mlszkg/Science-Design-System-Reference?node-id=642%3A513&scaling=min-zoom&page-id=642%3A0) (sizing names will probably have to be updated to fit our naming standards, I'm not sure where we are with that. If IconSize fits we could go with that). 

We've defined 4 sizes that correspond to the small or large (visual) style icons. Small-style icons can be icon-s or icon-xs, Large-style icons can be icon-l or icon-lx. 

Not sure if we want to implement this as a part of this ticket or if I should make a new one to implement icon sizes?

Reply:
Ooh fantastic! Thanks so much for the link, @Janeece !

Ah @j-x-han added a note in the PR description that we used medium instead of large, because MUI only has `small` and `medium` for the prop `size`. But since we actually have 4 sizes, what about `sdsSize="xs . s,  l,  xl` to differentiate from MUI's `size` props? 

Which also means... technically we could also prefix SDS's usage of `style` to be `sdsStyle` instead of using `phenotype` 😆  Am I opening a can of worms here lol??

CC: @ehoops @happyimadesignr @hthomas-czi 😂

Reply:
Thanks for the suggestions + link, I've added the four `iconSizes` to the theme!

I believe we only have 2 sizes for `IconButton` specifically (`small`/`large` in Figma, currently `small`/`medium` in this PR to match with MUI and our `Chip` component. These actually correspond to sizes `icon-s` and `icon-xl`).

Do we anticipate adding the other sizes for `IconButton`? If not, then I don't think it makes sense to change the `IconButton` prop to have all four sizes as options.

I am open to renaming the prop to `sdsStyle` or something similar, though! That could potentially help with the Figma small/large -> MUI small/medium discrepancy we're running into with some of these components.

Reply:
If there are no other objections/questions, I'll leave this as using `size?: "small" | "medium"` to stay consistent with MUI and our Chip component!